### PR TITLE
[MERL-132] Update eslint and related deps to v8

### PR DESCRIPTION
**Pull request by theodesp**
Signed-off-by: theodesp <theo.despoudis@wpengine.com>

## Description
This PR:

 - Updates eslint in all packages to v8( core, next, react)
 - Fixes eslint react hook errors and optional chaining warnings. The changes have no real side effects.

## Testing

Perform some smoke tests in the example project. It should load pages and posts correctly.

### CODE REVIEW

File: packages/next/src/server/getProps.tsx [hunk 1]
```diff
@@ -78,8 +78,9 @@ export async function getProps<
       async () => {
         const { cacheSnapshot: coreSnapshot } = await client.prepareReactRender(
           <RouterContext.Provider
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any,react/jsx-no-constructed-context-values
```
Comment (theodesp) [inline packages/next/src/server/getProps.tsx:81]: There shouldn't be a need to use `useMemo` here as this call happens on the server and this hook is not available to use.

File: packages/next/src/components/FaustProvider.tsx [hunk 2]
```diff
@@ -38,12 +38,14 @@ export function FaustProvider<Props = Record<string, unknown>>({
     cacheSnapshot: isNil(authSnapshot) ? undefined : authSnapshot,
   });
 
+  const value = React.useMemo(
```
Comment (blakewilson) [inline packages/next/src/components/FaustProvider.tsx:41]: Will introducing this have any side effects?

Comment (theodesp) [reply packages/next/src/components/FaustProvider.tsx:41]: No this is the recommended approach. The provider is dependant on the client which should be immutable object between each renders.

**Comment by theodesp** [comment]
> @theodesp Looks good! This techinically is a code change. I wonder if we should add a changeset?

Sure I will add one.

**Pull request merged**